projects/super_res/super_res.py

```diff
@@ -0,0 +1,203 @@
+import xarray as xr
+import numpy as np
+import fv3viz
+from vcm.catalog import catalog
+import matplotlib.pyplot as plt
+import os
+import cartopy.crs as ccrs
+
+GRID = catalog["grid/c384"].read()
+
+DPI = 800
+
+
+def repeat_arr(arr, upscale_factor):
+    # This repeats values along the last dimension
+    return np.repeat(arr, upscale_factor, axis=-1)
+
+
+def upscale_nearest(ds, upscale_factor, dim):
+    ds_upscaled = xr.apply_ufunc(
+        repeat_arr,
+        ds,
+        input_core_dims=[[dim]],  # apply function along this dimension
+        output_core_dims=[[dim]],  # the output will also have this dimension
+        kwargs={
+            "upscale_factor": upscale_factor
+        },  # pass in other arguments to the function
+        dask="parallelized",  # needed for dask arrays
+        output_sizes={
+            dim: ds.sizes[dim] * upscale_factor
+        },  # the size of the output dimension
+        exclude_dims=set((dim,)),  # allow changing the dimension size
+    )
+    return ds_upscaled
+
+
+def plot_pattern_bias(output: xr.Dataset, target: xr.Dataset, varname: str):
+    """Plot the pattern bias between two datasets
+
+    Args:
+        output (xr.Dataset): output dataset
+        target (xr.Dataset): target dataset
+    """
+    # compute the pattern bias
+    out_mean: xr.Dataset = output.mean("time")
+    target_mean: xr.Dataset = target.mean("time")
+    vmin = min(
+        # must convert min() output dask array to float explicitly,
+        # or we get an exception when it's used in plot_cube
+        out_mean[varname].min().values.item(),
+        target_mean[varname].min().values.item(),
+    )
+    vmax = max(
+        out_mean[varname].max().values.item(), target_mean[varname].max().values.item(),
+    )
+    bias: xr.Dataset = out_mean - target_mean
+    fig, ax = plt.subplots(
+        3, 1, figsize=(6, 9), subplot_kw={"projection": ccrs.Robinson()}
+    )
+    # plot the pattern bias
+    fv3viz.plot_cube(
+        out_mean.merge(GRID, compat="override"),
+        var_name=varname,
+        ax=ax[0],
+        vmin=vmin,
+        vmax=vmax,
+    )
+    ax[0].set_title("Time-mean Output")
+    fv3viz.plot_cube(
+        target_mean.merge(GRID, compat="override"),
+        var_name=varname,
+        ax=ax[1],
+        vmin=vmin,
+        vmax=vmax,
+    )
+    ax[1].set_title("Time-mean Target")
+    fv3viz.plot_cube(
+        # must merge grid data into the dataset
+        # after performing bias calculation on dataset
+        bias.merge(GRID, compat="override"),
+        var_name=varname,
+        cmap="RdBu_r",
+        ax=ax[2],
+    )
+    # TODO: add area-weighting to bias calculation
```

Comment:
Interesting, I guess this isn't terribly important at the moment?

Reply:
Not really, I left this out as it's a feature @cynicalsavant can add. It doesn't really take away from being able to use the plots to demonstrate the science.